cmd/backendapi/Dockerfile

```diff
@@ -1,14 +1,6 @@
-# Golang application builder steps
-FROM golang:1.11.5 as builder
-WORKDIR /go/src/github.com/GoogleCloudPlatform/open-match
-# Get only the necessary files from the repo.
-# https://stackoverflow.com/a/39317180/3113674
-# TODO: Set these to the final branch name after RC approval.
-RUN svn export https://github.com/GoogleCloudPlatform/open-match/branches/040wip/internal
-RUN svn export https://github.com/GoogleCloudPlatform/open-match/branches/040wip/config
-WORKDIR /go/src/github.com/GoogleCloudPlatform/open-match/cmd/backendapi
-COPY . .
-RUN go get -d -v
+FROM open-match-base-build as builder
+
+WORKDIR /go/src/github.com/GoogleCloudPlatform/open-match/cmd/backendapi/
 RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo .
 
```

Comment:
When developing locally, is there any way to avoid re-building of two images (base & cmd) without interventions to Dockerfile? Because all the logic is under `/internal/` now...

Reply:
When vendoring/go mod is introduced we can avoid it because the base image is just initializing and downloading the dependencies. And Open Match has a lot of dependencies. If you look at the base Dockerfile that's all it's doing.

Pulling the dependencies for each image (there's 10 of them) in Cloud Build takes so long that it was timing out (10 minutes). It was also flaky because the script was downloading all the dependencies for Open Match at one point more then 20 times! (10 docker, 10 binaries, 1 for test).

One thing I don't think we can get rid of is where the context of the Dockerfile's build starts. It'll need to start at the root of the repository, not at the main file. The reason is there's the dependencies, the main file, and files in other directories like internal/, and later on pkg/.

Reply:
I forgot to mention there is a way to build the Dockerfile without the whole repository context. You can prebuild the binary first and just copy that over so there's no building in the Dockerfile. This is pretty common practice as well and might be better since it aligns well with Skaffold.

I do have a branch where I take this approach but it's a major change that would take quite a few PRs and some buy in to get in. Example: https://github.com/jeremyje/open-match/blob/buildsys/cmd/backendapi/Dockerfile

Reply:
Moving this discussion to a proposal bug, https://github.com/GoogleCloudPlatform/open-match/issues/111

Reply:
If we want to revisit this we can but I'm going to submit the PR since we don't have any coverage at all for image builds.